### ISSUE TITLE
Words cut off after subscript (CO2-Einsparung becomes CO[newline])

### ISSUE THREAD

**Issue by wiertz** [open-discourse/open-discourse#100]
Thanks a lot for realizing this fantastic project! I found this error when browsing through speeches on climate and energy. See the results of this sql select:

```
next=# select id, document_url, (regexp_match(speech_content, '.{40}CO.{40}'))[1] 
next-# from open_discourse.speeches 
next-# where speech_content ~ 'CO\n'
next-# limit 3;
   id    |                   document_url                    |                regexp_match                
---------+---------------------------------------------------+--------------------------------------------
 1028335 | https://dip21.bundestag.de/dip21/btp/19/19115.pdf | . Dann kommt es zu einer Bepreisung von CO+
         |                                                   |                                           +
         |                                                   | ({4})                                     +
         |                                                   |                                           +
         |                                                   | Das ist ein Gesamtkonzept, das 
 1028251 | https://dip21.bundestag.de/dip21/btp/19/19114.pdf | , festzustellen, bei welchem Betrag ein CO+
         |                                                   |                                           +
         |                                                   | ({0})                                     +
         |                                                   |                                           +
         |                                                   | Der kann sehr hoch sein, und da
 1028253 | https://dip21.bundestag.de/dip21/btp/19/19114.pdf | ahr wird berichtet werden, wie sich der CO+
         |                                                   |                                           +
         |                                                   | Ich bin fest davon überzeugt, dass das
(3 rows)

next=# 
```

Seems to be some parsing issue, potentially happening if the subscript appears before a linebreak or an interjection?

**Comment by JabobKrauskopf** [comment]
Thanks for bringing this to our attention!

I just took a quick look at the XML-Structure where the error happened
```xml
Dann kommt es zu einer Bepreisung von CO<sub>2</sub> im nationalen Emissionshandel.
```
I think the `<sub>2</sub>` is messing up the extraction logic somehow. The 19th and 20th election period logic needs a rework anyway, which will probably happen in #81.
For now I'll probably add the quick and dirty fix of just removing all `<sub>`s and `</sub>`s - I think this should work and also have no side effects.

If this fix works then this issue will be taken care of in #99.

**Comment by wiertz** [comment]
Thanks for the quick response! A hunch from my brief attempt to understand the code:

https://github.com/open-discourse/open-discourse/blob/faa39e02ea1d315256b5c5b58601ca93587f718d/python/src/od_lib/05_electoral_term_19/01_extract_speeches_and_contributions_electoral_term_19.py#L368-L370

This is where text is extracted from `<p>` tags in the speech content. However, content.text only parses the top level text. If it is conceivable that there are other nested tags, then it might make sense to use itertext() instead:


```
>>> import xml.etree.ElementTree as et
>>> xmlstring = '<p>A lot of CO<sub>2</sub> is in the atmosphere.</p>'
>>> content = et.fromstring(xmlstring)
>>> 
>>> print(content.text)
A lot of CO
>>> 
>>> print(''.join(content.itertext()))
A lot of CO2 is in the atmosphere.
>>> 
>>> 
```

**Comment by JabobKrauskopf** [comment]
Nice catch! Thanks for pointing that out! That definitely is the problem.

However I just checked for unintended side-effects when using `''.join(content.itertext())` and so far I have found around 5.000 instances which is quite a lot for only 277 session, so for now I'll still only implement the quick and dirty fix and use the `''.join(content.itertext())` solution in #81 when reworking the whole logic.
